### PR TITLE
Testing access

### PR DESCRIPTION
**Pull request by EclipseGc**
This looks like a huge improvement to me. It even denied my anonymous attempt to delete a block, so definitely an improvement.

### CODE REVIEW

File: src/Controller/Services.php [hunk 1]
```diff
@@ -66,11 +66,12 @@ public function processRequest(Request $request, RouteMatchInterface $route_matc
      */
     foreach ($service_def->getContextDefinitions() as $context_id => $context_definition) {
       if ($request->attributes->has($context_id)) {
-        $context = new Context($context_definition);
-        $context = Context::createFromContext($context, $request->attributes->get($context_id));
+        $context = new Context($context_definition, $request->attributes->get($context_id));
+        // $context = Context::createFromContext($context, $request->attributes->get($context_id));
```
Comment (nerdstein) [inline src/Controller/Services.php]: Are we planning to use this moving forward? If not, we should just remove it

File: src/Plugin/Deriver/EntityDelete.php [hunk 2]
```diff
@@ -17,9 +17,9 @@ public function getDerivativeDefinitions($base_plugin_definition) {
       $this->derivatives[$entity_type_id]['title'] = $this->t('@label: Delete', ['@label' => $entity_type->getLabel()]);
       $this->derivatives[$entity_type_id]['description'] = $this->t('Deletes a @entity_type_id object.', ['@entity_type_id' => $entity_type_id]);
       $this->derivatives[$entity_type_id]['category'] = $this->t('@label', ['@label' => $entity_type->getLabel()]);
-      $this->derivatives[$entity_type_id]['path'] = "$entity_type_id/{entity}";
+      $this->derivatives[$entity_type_id]['path'] = "$entity_type_id/{{$entity_type_id}}";
       $this->derivatives[$entity_type_id]['context'] = [
-        'entity' => new ContextDefinition("entity:$entity_type_id", $this->t('@label', ['@label' => $entity_type->getLabel()]))
+        "$entity_type_id" => new ContextDefinition("entity:$entity_type_id", $this->t('@label', ['@label' => $entity_type->getLabel()]))
```
Comment (nerdstein) [inline src/Plugin/Deriver/EntityDelete.php:22]: I may be crazy but I thought the RC version of core uses just "label()" now. Am I mistaken?

File: src/Plugin/ServiceDefinition/EntityDelete.php [hunk 3]
```diff
@@ -28,9 +29,16 @@ class EntityDelete extends ServiceDefinitionBase {
   /**
    * {@inheritdoc}
    */
+  public function processRoute(Route $route) {
+    $route->setRequirement('_entity_access', $this->getDerivativeId() . '.delete');
```
Comment (nerdstein) [inline src/Plugin/ServiceDefinition/EntityDelete.php:33]: I am curious what getDerivativeID is providing for us here? Is this some token or something?

Comment (kylebrowning) [inline src/Plugin/ServiceDefinition/EntityDelete.php:33]: It gets us the type of entity were using.

So block, or taxonomy, etc ,etc

File: src/Controller/Services.php (hunk 1 above)
Comment (EclipseGc) [inline src/Controller/Services.php]: We actively use context already in most of the plugins.

Comment (kylebrowning) [inline src/Controller/Services.php]: I think he's referring to the commented code.

**Pull request merged**